modules/storages/app/common/storages/peripherals/storage_interaction/one_drive/create_folder_command.rb

```diff
@@ -97,15 +100,15 @@ def file_info_for(json_file)
             )
           end
 
-          def payload(folder_path)
+          def payload(folder_name)
             {
-              name: folder_path,
+              name: folder_name,
               folder: {},
               "@microsoft.graph.conflictBehavior" => "fail"
             }.to_json
           end
 
-          def base_uri = "/v1.0/drives/#{@storage.drive_id}"
+          def base_uri = Util.join_uri_path(@storage.uri, "v1.0", "drives", @storage.drive_id)
```

Comment:
🟡 Do we need to use `File.join` (knowing it will break on Windows) at all? We know how to build the URI and hardly it has any data coming from a user, so not even that would break it.

Reply:
The main reason for using it is to harden the usage against trailing slashes in the data. Is Windows really an issue here? Windows is no target platform for OP.

Reply:
It is more about the signal vs noise ratio here. The URI is fixed and has no trailing slash, and if configured correctly it won´t have one or the entire integration won't work.

This is a issue on Nextcloud as there's user define path parts (Folder names). As for file names, OD don't even allow slashes (of either type).